ext/nokogiri/extconf.rb

```diff
@@ -468,7 +468,7 @@ def install
 
   unless windows_p
     preserving_globals {
-      have_library('z', 'gzdopen', 'zlib.h')
+      have_library('z', 'gzdopen')
```

Comment:
Should be `have_library('z', 'gzdopen') || have_library('z', 'gzdopen', 'zlib.h')`.  There might be a platform where gzdopen() is a macro defined in zlib.h.